local_db/data_access.py

```diff
@@ -99,6 +99,13 @@ def get_requests_for_workspace(self, workspace: str):
             )
         ]
 
+    def get_released_files_for_workspace(self, workspace: str):
+        return dict(
+            RequestFileMetadata.objects.filter(
+                request__workspace=workspace, released_at__isnull=False
+            ).values_list("relpath", "file_id")
```

Comment:
Question using relpath as the dict key here - what happens if the same relpath has been released twice?  Are we relying on the the default query order to mean the latest released file will win the key race?  I think this works, as the version on disk should be the latest?

If so, should we a) explicitly order the query to ensure this and b) have a comment explaining why?

Is there a way this could work if file_id was the key, which might avoid this particular issue? Or is that not actually what we want to look up on?

Reply:
Mmm true, it could be a list of previously released file ids instead. I suppose you could have released one version, changed it and changed it back again. At the time, I was thinking that since we don't allow releasing more than once, there'd only ever be one.
I think it could use the file_id as the key, that's probably the best way

Reply:
Ah, if I use file_id it just needs to be a list of file ids, not relpaths. Then you also can't re-release the same file under a different path/name (which means a few more updates to tests).

Reply:
I think this is what we want, because that's what job-server will refuse to release, right?